ostd/src/io/io_mem/allocator.rs

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! I/O Memory allocator.
+
+use alloc::vec::Vec;
+use core::ops::Range;
+
+use log::{debug, info};
+use spin::Once;
+
+use crate::{
+    io::io_mem::IoMem,
+    mm::{CachePolicy, PageFlags},
+    util::range_alloc::RangeAllocator,
+};
+
+/// I/O memory allocator that allocates memory I/O access to device drivers.
+pub struct IoMemAllocator {
+    allocators: Vec<RangeAllocator>,
+}
+
+impl IoMemAllocator {
+    /// Gets the I/O memory access for `range`.
+    ///
+    /// If the range is not available, then the return value will be `None`.
+    pub fn get(&self, range: Range<usize>) -> Option<IoMem> {
```

Comment:
Rename this method to `acquire` to convey the meaning that this method is more heavy-weight that one normally expects for a `get` method.